pkg/daemon/daemon.go

```diff
@@ -539,8 +540,25 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 	}
 
 	if reqReboot {
-		glog.Info("nodeStateSyncHandler(): reboot node")
-		rebootNode()
+		if err := wait.PollImmediateUntil(3*time.Second, func() (bool, error) {
+			if utils.ClusterType == utils.ClusterTypeOpenshift {
+				cmc, _ := dn.node.Annotations[mcdconst.CurrentMachineConfigAnnotationKey]
+				dmc, _ := dn.node.Annotations[mcdconst.DesiredMachineConfigAnnotationKey]
+				mcdState, _ := dn.node.Annotations[mcdconst.MachineConfigDaemonStateAnnotationKey]
+				if cmc == "" || cmc != dmc || mcdState != mcdconst.MachineConfigDaemonStateDone {
+					glog.Info("nodeStateSyncHandler(): node is not ready for reboot, wating for MCO state Done")
+					return false, nil
+				}
+			}
+
+			glog.Info("nodeStateSyncHandler(): reboot node")
+			rebootNode()
```

Comment:
If MCO happens to apply kernel arguments while SR-IOV requesting reboot for its iommu change, will SR-IOV's change be override by MCO's change?

Reply:
Will MCO become degraded when other components applying kernel argument change?

Reply:
> If MCO happens to apply kernel arguments while SR-IOV requesting reboot for its iommu change, will SR-IOV's change be override by MCO's change?
I thought we were pausing pools, such that the MCO would operate after SRIOV completes. In that case the MCO wouldn't touch any kargs it doesn't have listed.

> Will MCO become degraded when other components applying kernel argument change?
The MCO does not validate kargs I believe, so it won't degrade.

Reply:
The pausing MCO approach requires SNO to make a big amount of change. It's hard to coordinate the pause/resume operator among different nodes. There is a big difference between MCO and SRO is that the controller of SRO has no control over the order of the node reboot.

Reply:
> The pausing MCO approach requires SNO to make a big amount of change. It's hard to coordinate the pause/resume operator among different nodes. There is a big difference between MCO and SRO is that the controller of SRO has no control over the order of the node reboot.

Does the controller know when the nodes are done? Would the controller be able to pause the pool, let the nodes do their thing, and then unpause once the nodes finish? Or does the controller not have insight into when the nodes are done?

Reply:
No, the controller doesn't know when the nodes are done. It only tells the daemon the desired state of the node. The config daemon will update the status of the node SriovNetworkNodeState CR, when it's done. But the controller doesn't check that.